cradlepoint/util/addEntry.js

```diff
@@ -0,0 +1,34 @@
+import {testSchema} from "../schemas/testSchema";
+import connectToDb from "./mongodb";
+const { ObjectId } = require('mongodb');
+
+export const testMessage = 'test'
+
+export async function addTest(data) {
+    try {
+        const client = await connectToDb();
+        const valid = await testSchema.isValid(data)
+        if (valid && ObjectId.isValid(data.testCaseId) ) {
+            const test = testSchema.cast(data);
+            for (const result in test.results) {
```

Comment:
Replace the for loop with something like this:
const validResults = !data.results.map((str) =>ObjectId.isValid(str)).includes(false);

Reply:
I don't think that's better. This solution only loops through a single time instead of twice and sends an error if we ever have an invalid ID. Mapping with this would require another check